src/MongoDB.Driver/IndexKeysDefinitionBuilder.cs

```diff
@@ -414,6 +414,30 @@ public IndexKeysDefinition<TDocument> Text(Expression<Func<TDocument, object>> f
         {
             return Text(new ExpressionFieldDefinition<TDocument>(field));
         }
+
+        /// <summary>
+        /// Creates a wildcard index key definition.
+        /// </summary>
+        /// <param name="field">The wildcard key.</param>
+        /// <returns>A wildcard index key definition.</returns>
+        public IndexKeysDefinition<TDocument> Wildcard(FieldDefinition<TDocument> field = null)
+        {
+            if (field == null)
+            {
+                field = "$**";
+            }
+            return new WildcardIndexKeyDefinition<TDocument>(field);
+        }
+
+        /// <summary>
+        /// Creates a wildcard index key definition.
+        /// </summary>
+        /// <param name="field">The wildcard key.</param>
+        /// <returns>A wildcard index key definition.</returns>
+        public IndexKeysDefinition<TDocument> Wildcard(Expression<Func<TDocument, object>> field)
+        {
+            return Wildcard(new ExpressionFieldDefinition<TDocument>(field));
+        }
```

Comment:
Note: the following call to Wildcard is ambiguous:

    var keys = Builders<C>.IndexKeys.Wildcard(null);

Probably doesn't matter. I think there are similar ambiguities in lots of places.